blink/syscall.c

```diff
@@ -1542,7 +1552,35 @@ static i64 SysGetdents(struct Machine *m, i32 fildes, i64 addr, i64 size) {
       reclen = 0;
       continue;
     }
+#ifdef DT_UNKNOWN
     type = UnXlatDt(ent->d_type);
+#else
+    struct stat st;
+    if (fstatat(fd->fildes, ent->d_name, &st, AT_SYMLINK_NOFOLLOW) == -1) {
```

Comment:
Nice. We may want to do this on other platforms too. For example, some filesystems on Linux (like ReiserFS or BtrFS I think) don't support `DT_*` constants and I think they set it to `DT_UNKNOWN` when that happens. I think it might be valuable to try to obtain that information from `fstatat()` when that happens.